### PR TITLE
pass dropped links to the urlbar, not the browser

### LINKED ISSUE #1671: Bug : Brave Suddenly shows full white page
Hello Dev Team,

The browser suddenly shows full white blank page and all tabs and toolbars are not accessible.
This happened when i was accidently moving a hyperlink.

OS : windows 7 64 bit (SP1)
Brave : 0.9.4

As you can see below:

bravesuddenlystopped

### LINKED ISSUE #1891: drag-and-drop bookmarklet to navigation bar causes whitescreen
**Did you search for similar issues before submitting this one?**
yes

**Describe the issue you encountered:**
dragging and dropping a javascript: url to the navigation bar area (not the bookmarks toolbar) causes the browser to whitescreen. demo: open https://jsfiddle.net/r2ftho8k/ and drag the 'click me' link to the navbar.

**Expected behavior:**
nothing should happen

- Any related issues:
may be a regression of https://github.com/brave/browser-laptop/issues/259

### PR DESCRIPTION
**Pull request by bridiver**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Ran `git rebase -i` to squash commits if needed.

fixes #1671
fixes #1891 

@diracdeltas @bbondy

### CODE REVIEW

File: js/components/bookmarksToolbar.js [hunk 1]
```diff
@@ -215,19 +215,18 @@ class BookmarksToolbar extends ImmutableComponent {
       if (a && a.href) {
         appActions.addSite({
           title: a.innerText,
-          location: a.href
+          location: e.dataTransfer.getData('text/plain')
```
Comment (bridiver) [inline js/components/bookmarksToolbar.js]: a.href was truncated sometimes and I'm not sure why, but text/plain always contained the correct link text

File: js/components/main.js [hunk 2]
```diff
@@ -424,16 +445,20 @@ class Main extends ImmutableComponent {
...
     }
   }
 
   onDrop (e) {
-    if (e.dataTransfer.files) {
+    if (e.dataTransfer.files.length > 0) {
       Array.from(e.dataTransfer.files).forEach((file) =>
         windowActions.newFrame({location: file.path, title: file.name}))
+    } else if (e.dataTransfer.getData('text/plain')) {
+      windowActions.setNavBarUserInput(e.dataTransfer.getData('text/plain'))
```
Comment (bridiver) [inline js/components/main.js]: right now this just loads the url into the urlbar. Do we want it to actually load the url?

File: js/components/bookmarksToolbar.js (hunk 1 above)
Comment (bbondy) [inline js/components/bookmarksToolbar.js]: ok thanks, was wondering about this.

File: js/components/main.js (hunk 2 above)
Comment (bbondy) [inline js/components/main.js]: yes

Comment (bbondy) [inline js/components/main.js]: load the url

**Comment by bbondy** [comment]
++

**Pull request merged**